### ISSUE TITLE
Keyboard is not working, but I can paste

### ISSUE THREAD

**Issue by donquixote** [tomboy-notes/tomboy-ng#315]
Platform:
KDE Plasma on Ubuntu (more below).

Steps:
1. Open or create a note.
2. Put the caret into the text box.
3. Start typing.
4. Copy a text from elsewhere, and paste it into the text box, using the mouse.

Expected:
- On 3., the typed letters appear.
- On 4., the text from copy/paste appears.

Actual:
- (bad) On 3, typed letters do not appear in the note.
- (bad) Also, other shortcuts like Ctrl-Z or backspace don't work.
- (good) The text from copy/paste appears.
- (good) I can move the text caret with the mouse.

More observations:
- In the past it used to work. Something must have changed.
- Other applications receive key strokes just fine.

Conclusions:
- The note is not "write-only" (if that is even possible).
- Key strokes don't make it into the note app.

More platform info:

> Operating System: Ubuntu 12
> KDE Plasma Version: 5.27.11
> KDE Frameworks Version: 5.115.0
> Qt Version: 5.15.13
> Kernel Version: 6.8.0-41-generic (64-bit)
> Graphics Platform: X11
> Processors: 8 × 11th Gen Intel® Core™ i7-1165G7 @ 2.80GHz
> Memory: 15,3 GiB of RAM
> Graphics Processor: Mesa Intel® Xe Graphics
> Manufacturer: LG Electronics
> Product Name: 17Z90P-G.AA79G
> System Version: 0.1

**Comment by davidbannon** [comment]
Don, you have absolutely stumped me with this one. I routinely test on KDE and never seen anything like this.  Some questions please -
* Your version of Ubuntu, '12' can you please check that, I doubt you mean 12.04, that  is twelve years old. I might have to build a replica.
* In tomboy-ng, can you type into other places, eg, the search field from the Search Window ?
* Which version of tomboy-ng are you using ?  One from Ubuntu repository or downloaded from here (.deb or .AppImage ?).

Davo

**Comment by donquixote** [comment]
Hi.
With `lsb_release -a' I see "Ubuntu 24.04.1 LTS".

But my "About this system" dialog says Ubuntu 12.
So this dialog is really malfunctioning.

Here the output from tomboy-ng "About".
(Unfortunately I don't find how to copy this so I have to type.)

> Version 0.39, QT5
> Build date 2024/04/01 (Laz 3.0.0)
> TargetCPU x86_64 OS Linux KDE
> QT_QPA_PATFORMTHEME : 
> QT_QPA_PLATFORM : xcb

In synaptic package manager it says tomboy-ng 0.39-1build2
I suppose this means I did use that for installing it.

**Comment by donquixote** [comment]
> In tomboy-ng, can you type into other places, eg, the search field from the Search Window ?

Good that you ask!
I can use the keyboard in the main dialog window with the search box etc.
I can _not_ use the keyboard in the window that shows an individual note.

**Comment by davidbannon** [comment]
OK, I am pulling down a KUbuntu 24.04-1 image as we speak. Pretty slow network here ...

Yeah, their "About this system" is broken !

tomboy-ng 0.39-1build2 sounds like an Ubuntu rebuild of a Debian package, nothing unusual about that.

And only not working in the note window ? That sounds like a KMemo issue to me and that could be quite messy. You are already using xcb so blaming Wayland is unlikely but not impossible. 

So, in case I cannot replicate it here, some tests please ?
Could you please log out (not shutdown) and select your user name but before typing in your password, switch to X11 instead of Wayland. From memory, a clickable link lower left of the logon screen.  This is quite reversible, nothing permanent.

Then, password and test tomboy-ng again.  I have found some substantial differences in how KDE adapts to Wayland (compared to Gnome) so, its possible. 

Davo

**Comment by davidbannon** [comment]
Yeah, I am afraid I cannot replicate that problem. And everything matches. Its even tells me its running Ubuntu 12 too !

But no sign of that problem.

Now Don, that means if we are to get to the bottom of this, it might need some help from your end. Not sure exactly what just yet ....

Davo

**Comment by donquixote** [comment]
Alright, thanks for the quick feedback!
I can test with and without wayland, sure.
Or reinstall tomboy-ng. Or reinstall my system which seems a bit unreliable lately.

If nothing else, perhaps if others have the same experience they can post here.

**Comment by davidbannon** [comment]
Hmm, not a big fan of reinstalling, this is not Windows.  tomboy-ng is pretty deterministic, all its settings are visible to you, nothing happening below your gaze. I'd like to think your OS is too. Maybe not.

But do look in your settings.
* Settings -> Input Methods : you get a list of characteristics, be good to compare to mine, might be a screen scrape ...
* Settings -> : regional settings perhaps ?

Make sure all your work is saved before you start this, I managed to lock KUbuntu up by looking at the regional settings page. Thats a most un-ubuntu sort of thing. Maybe some interaction with VirtualBox ?

Now, other things to try. Get these going, try to type into a note window .... quit.

First, easy one, please start tomboy-ng from the command line, useful to see if there are any debug messages coming through.

```
$> tomboy-ng  <enter>
...
$> QT_QPA_PLATFORM=xcb  tomboy-ng <enter>
```

In both cases, please ensure that your tomboy-ng is not running before you start, if it is, it just gets brought forward and the new one you are trying to start, terminates.

Anyway, another thing I'd like you to try please.  In the download repo, here,  https://github.com/tomboy-notes/tomboy-ng/releases/tag/v0.40a there are some beta test install kits. No, I don't expect a new .deb will help, I have made no recent (or otherwise) changes to input tools. But there are two AppImage packages there, recognizable by their name. One for Qt5 and one for Qt6, they both should run without any further libraries or installing. Just download, set the file's attribute to executable  ` $>  chmod u+x  tomboy-ng-x86_64-Qt5.AppImage  <enter>`  and make sure your existing tomboy-ng is not running, then run it. The Qt6 one is particularly interesting as its possible (just) that we are seeing a Qt problem. 

Thanks for your help !

Davo

**Comment by donquixote** [comment]
I have interesting observations.

On wayland it works perfectly.
On X11 it does not (even after restart).

But I notice that on X11 I can use German umlaut keys ö, ä, ü.
However I cannot use any of the other letters or keys like backspace etc.

**Comment by davidbannon** [comment]
OK, now, that is confronting !  It works on Wayland but not (generally) on X11 ?   Thats the very first time I heard of anything working better under Wayland !

Did you try the comman line tests ?  A Qt5 app uses Wayland directly unless the QT_QPA_PLATFORM is set to xcb.  I wonder if your issue is xcb related ?

Using the GTK2 version of tomboy-ng suddenly becomes interesting too. It ignores Wayland completely going direct to X11.

Davo
